trino-aws-proxy/src/main/java/io/trino/aws/proxy/server/rest/RequestBuilder.java

```diff
@@ -50,19 +53,33 @@ private RequestBuilder() {}
     static Request fromRequest(ContainerRequest request)
     {
         MultiMap requestHeaders = ImmutableMultiMap.copyOfCaseInsensitive(request.getHeaders().entrySet());
-        String xAmzDate = requestHeaders.getFirst("x-amz-date").orElseThrow(() -> {
-            log.debug("Missing \"x-amz-date\" header");
-            return new WebApplicationException(Response.Status.BAD_REQUEST);
-        });
-        Optional<String> securityTokenHeader = requestHeaders.getFirst("x-amz-security-token");
+        SigningQueryParameters.SplitQueryParameters queryParameters = splitQueryParameters(ImmutableMultiMap.copyOf(request.getUriInfo().getQueryParameters(true).entrySet()));
+        MultiMap authorizationQueryParameters = ImmutableMultiMap.copyOf(queryParameters.signingQueryParameters());
+        AwsTimestamp requestTimestamp = AwsTimestamp.fromRequestTimestamp(
+                requestHeaders.getFirst("x-amz-date")
+                    .or(() -> authorizationQueryParameters.getFirst("X-Amz-Date"))
```

Comment:
why not use the case-insensitive MultiMap?

Reply:
Query parameters are case sensitive (https://www.rfc-editor.org/rfc/rfc3986#section-6.2.2.1)